### PR TITLE
feat: Add support for watch flag

### LINKED ISSUE #27: Add support for the `--watch` flag
It would be great to be able to auto-run tests on save. Other adapters like `neotest-jest` and `neotest-python` support this. [The PR](https://github.com/nvim-neotest/neotest-jest/pull/50) that added support for this to `neotest-jest` looks like it should map pretty cleanly onto this repo as well. 

Resources:

- https://github.com/nvim-neotest/neotest-jest/pull/50
- https://github.com/nvim-neotest/neotest/discussions/20

### PR DESCRIPTION
**Pull request by marcoSven**
This commit introduces a feature to support the watch mode in neotest. In the previous implementation, the user had the ability to run tests but there was no provision to watch the files and run the tests when changes occur. This feature is particularly helpful while doing test-driven development where code changes frequently.

1. Added key mappings examples in README.md to trigger watch mode.
2. Refactored the build_spec function in init.lua to support watch mode. It now includes command flag "--watch=false" in the command list instead of --run.
3. Added a new function stream in util.lua which continuously reads file data and triggers a callback function for new data. This is used to watch the test files and run tests on file change.

https://github.com/marilari88/neotest-vitest/assets/3860298/649933e2-6944-4bf0-a4e8-f487159a2d90

closes #27

### CODE REVIEW

**Comment by marilari88** [comment]
@marcoSven  Big thanks for your contribution! I left a couple of comments for you. One concern though: I noticed when testing the PR that the vitest process in watch mode sticks around until Neovim shuts down or Neotest stop is called. This could lead to a bunch of unwanted background processes if we're running lots of tests in watch mode. How do you handle it?

File: lua/neotest-vitest/init.lua [hunk 1]
```diff
@@ -225,6 +225,7 @@ local function get_strategy_config(strategy, command)
         runtimeExecutable = command[1],
         console = "integratedTerminal",
         internalConsoleOptions = "neverOpen",
+        cwd = cwd or "${workspaceFolder}",
```
[the remark this answers is not in the document]
Comment (marcoSven) [reply lua/neotest-vitest/init.lua:228]: Yes, this is not related to the feature. This a requirement for how I configure my debug. 

This option is also provided by Neotest Jest https://github.com/nvim-neotest/neotest-jest/blob/959d45b133de938c79e3f064db188680eaf69055/lua/neotest-jest/init.lua#L272

**Comment by marcoSven** [comment]
> I noticed when testing the PR that the vitest process in watch mode sticks around until Neovim shuts down or Neotest stop is called. 

I think this is the same with Jest adapter.

**Comment by marcoSven** [comment]
I noticed when testing the PR that the vitest process in watch mode sticks around until Neovim shuts down or Neotest stop is called.

I think this is the same with the Jest adapter.

**Comment by marcoSven** [comment]
> I noticed when testing the PR that the vitest process in watch mode sticks around until Neovim shuts down or Neotest stop is called.

I think this is the same with the Jest adapter.

**Comment by marilari88** [comment]
@marcoSven thanks!

**Pull request merged**